static/src/javascripts/projects/common/modules/on-images-loaded-fixed.js

```diff
@@ -0,0 +1,27 @@
+import { memoize } from 'lodash-es';
+
+const getFuncId = (rules) => rules.bodySelector || 'document';
+
+const query = (selector, context) => [
+	...(context ?? document).querySelectorAll(selector),
+];
+
+export const onImagesLoadedFixed = memoize((rules) => {
+	const notLoaded = query('img', rules.body).filter(
+		(img) => !img.complete && img.loading !== 'lazy',
+	);
+
+	return notLoaded.length === 0
+		? Promise.resolve()
+		: new Promise((resolve) => {
+				let loadedCount = 0;
+				notLoaded.forEach((img) =>
+					img.addEventListener('load', () => {
+						loadedCount += 1;
+						if (loadedCount === notLoaded.length) {
+							resolve();
+						}
+					}),
+				);
+		  });
```

Comment:
non-blocking: I know this is just fixing the bug in the original code but it might be good to simplify this logic a bit while we're here. I find the following easier to parse but feel free to leave it as-is if you prefer:
```suggestion
	const imgPromises = notLoaded.map(
		(img) =>
			new Promise((resolve) => {
				img.addEventListener('load', resolve);
			}),
	);
	return Promise.all(imgPromises);
```

Reply:
Great suggestion @zekehuntergreen, that's miles better. Have made the change